tooling/noir_codegen/src/index.ts

```diff
@@ -1,24 +1,53 @@
 import { CompiledCircuit } from '@noir-lang/types';
+import { PrimitiveTypesUsed, generateTsInterface } from './noir_types.js';
 
-const codegenImports = `import { InputMap, InputValue } from "@noir-lang/noirc_abi"
-import { Noir } from "@noir-lang/noir_js"`;
+// TODO: reenable this. See `abiTypeToTs` for reasoning.
+// export type FixedLengthArray<T, L extends number> = L extends 0 ? never[]: T[] & { length: L };
```

Comment:
Comment from call: Ad tracking issue for this, we use it because when we have an incorrect array type, the error does not tell us exactly which array has too many/few items. If we could narrow down to the field in the type instead of the parameter in the main function we would never need this struct

Reply:
https://github.com/noir-lang/noir/issues/3560